File: csharp/DesiArchitect.DesignPatterns/05_Singleton/After/OrderService.cs
namespace DesiArchitect.DesignPatterns.Singleton.After;

public class OrderService
{
    private readonly IAppConfig _config;

    public OrderService(IAppConfig config) => _config = config;

    public void PlaceOrder(string item)
    {
        Console.WriteLine($"  [After] OrderService using injected IAppConfig");
        Console.WriteLine($"     Placing order for {item} using key: {_config.ApiKey}");
        Console.WriteLine($"     Testable - inject a fake IAppConfig in unit tests");
    }
}

File: typescript/src/05-singleton/after/order-service.ts
import type { IAppConfig } from './i-app-config.js';

export class OrderService {
  constructor(private readonly config: IAppConfig) {}

  placeOrder(item: string): void {
    console.log('  [After] OrderService using injected IAppConfig');
    console.log(`     Placing order for ${item} using key: ${this.config.apiKey}`);
    console.log('     Testable - inject a fake IAppConfig in unit tests');
  }
}